lib/ansible/modules/network/ftd/ftd_install.py

```diff
@@ -0,0 +1,305 @@
+#!/usr/bin/python
+
+# Copyright (c) 2019 Cisco and/or its affiliates.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: ftd_install
+short_description: Installs FTD pkg image on the firewall
+description:
+  - Provisioning module for FTD devices that installs ROMMON image (if needed) and
+    FTD pkg image on the firewall.
+  - Can be used with `httpapi` and `local` connection types. The `httpapi` is preferred,
+    the `local` connection should be used only when the device cannot be accessed via
+    REST API.
+version_added: "2.8"
+requirements: [ "python >= 3.5", "kick" ]
```

Comment:
Does Ansible require support for Python 2.7 as well? If so, how would that impact this module?

Reply:
This module depends on the `unicon` library for connecting to the device and interacting with it. Unfortunately, `unicon` is available for Python 3.5+ only, so this module cannot be used with Python 2.7.
I checked that other modules (for example, [ios_static_route](https://github.com/ansible/ansible/blob/c8914a1678c8117304bba1a883dd37751a07f88b/lib/ansible/modules/network/ios/ios_static_route.py#L39) have a Python 3+ requirement, so put it here the same way for now.

Reply:
The `ios_static_route` module works with Python 2.7 version as well. It seems the `ipaddress` python module is shipped with Python 3.3 version onwards, that's the reason for the explicit mention in the requirement.

Minimum python version supported on the control node is 2.7 https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#control-machine-requirements

Generally, the module running on a control node is expected to support Python 2.7 version as well, not sure if currently there is an exception to this requirement.

Reply:
Please ignore my above comment. 
As per @abadger the syntax of the module should compatible with python-2.7 but can be documented as not running under other versions (when there is a necessary library which doesn't support it)

Also, @jborean93 pointed out that the module can run under a different python on localhost so it's still possible to run Ansible on 2.7 but use a python 3 interpreter for a specific module.
eg:
```
- mymodule:
  vars:
    ansible_python_interpreter: /usr/bin/python3
```

Reply:
Thanks for clear explanation, @ganeshrn. It makes perfect sense. 

The `ftd_install` module is already compatible with Python 2.7 and its unit tests pass with Python 2.7. I also added notes to `DOCUMENTATION` section telling that Python interpreter >= 3.5 is required for this particular module.